src/OpenLoco/Paint/Paint.cpp

```diff
@@ -336,6 +416,68 @@ namespace OpenLoco::Paint
         }
     }
 
+    void PaintSession::attachStringStruct(PaintStringStruct& psString)
+    {
+        auto* previous = *_lastPaintString;
+        _lastPaintString = &psString;
+        if (previous == nullptr)
+        {
+            _paintStringHead = &psString;
+        }
+        else
+        {
+            previous->next = &psString;
+        }
+    }
+
+    PaintStruct* PaintSession::createNormalPaintStruct(uint32_t imageId, const Map::Pos3& offset, const Map::Pos3& boundBoxOffset, const Map::Pos3& boundBoxSize)
+    {
+        auto* const g1 = Gfx::getG1Element(imageId);
+        if (g1 == nullptr)
+        {
+            return nullptr;
+        }
+
+        const auto swappedRotation = directionFlipXAxis(currentRotation);
+        auto swappedRotCoord = Map::Pos3{ Math::Vector::rotate(offset, swappedRotation), offset.z };
+        swappedRotCoord += Map::Pos3{ getSpritePosition(), 0 };
+
+        const auto vpPos = Map::gameToScreen(swappedRotCoord, currentRotation);
+
+        if (!imageWithinContext(vpPos, *g1, **_context))
+        {
+            return nullptr;
+        }
+
+        const auto rotBoundBoxOffset = Map::Pos3{ Math::Vector::rotate(boundBoxOffset, swappedRotation), boundBoxOffset.z };
+        const auto rotBoundBoxSize = rotateBoundBoxSize(boundBoxSize, currentRotation);
+
+        auto* ps = allocatePaintStruct<PaintStruct>();
+        if (ps == nullptr)
+        {
+            return nullptr;
+        }
+
+        ps->imageId = imageId;
+        ps->x = vpPos.x;
+        ps->y = vpPos.y;
+        ps->bounds.xEnd = rotBoundBoxSize.x + rotBoundBoxOffset.x + getSpritePosition().x;
+        ps->bounds.yEnd = rotBoundBoxSize.y + rotBoundBoxOffset.y + getSpritePosition().y;
+        ps->bounds.zEnd = rotBoundBoxSize.z + rotBoundBoxOffset.z;
+        ps->bounds.x = rotBoundBoxOffset.x + getSpritePosition().x;
+        ps->bounds.y = rotBoundBoxOffset.y + getSpritePosition().y;
+        ps->bounds.z = rotBoundBoxOffset.z;
+        ps->flags = 0;
+        ps->attachedPS = nullptr;
+        ps->children = nullptr;
+        ps->type = _itemType;
+        ps->var_29 = _trackModId;
```

Comment:
should we rename `var_29` to `trackModId`? or is var_29 used for other things?

Reply:
Like most things it has multiple uses but yeah should rename it to perhaps modId (it's not track specific) also need to rename unkBh